polus-feature-extraction-plugin/src/main.py

```diff
@@ -0,0 +1,1226 @@
+      
+    #Capitalize the first letter of header
+    df_insert.columns = map (lambda x: x.capitalize(), df_insert.columns)
+       
+    return df_insert, title
+
+# Setup the argument parsing
+def main():
+    logger.info("Parsing arguments...")
+    parser = argparse.ArgumentParser(prog='main', description='Everything you need to start a Feature Extraction plugin.')
+    parser.add_argument('--features', dest='features', type=str,
+                        help='Features to calculate', required=True)
+    parser.add_argument('--csvfile', dest='csvfile', type=str,
+                        help='Save csv as separate or single file', required=True)
+    parser.add_argument('--embeddedpixelsize', dest='embeddedpixelsize', type=str,
+                        help='Embedded pixel size if present', required=False)
+    parser.add_argument('--pixelsPerunit', dest='pixelsPerunit', type=float,
+                        help='Pixels per unit', required= False)
+    parser.add_argument('--unitLength', dest='unitLength', type=str,
+                        help='Required units for features extracted', required= False)
+    parser.add_argument('--intDir', dest='intDir', type=str,
+                        help='Intensity image collection', required=False)
+    parser.add_argument('--pixelDistance', dest='pixelDistance', type=int,
+                        help='Pixel distance to calculate the neighbors touching cells', required=False)
+    parser.add_argument('--segDir', dest='segDir', type=str,
+                        help='Segmented image collection', required=True)
+    parser.add_argument('--outDir', dest='outDir', type=str,
+                        help='Output collection', required=True)
+
+    # Parse the arguments
+    args = parser.parse_args()
+    
+    #List of features to be extracted
+    features = args.features.split(',')
+    logger.info('features = {}'.format(features))
+    
+    #Save the features extracted (as single file for all images or 1 file for each image) in csvfile
+    csvfile = args.csvfile
+    logger.info('csvfile = {}'.format(csvfile))
+    
+    #Embedded pixel size if true, get units from metadata
+    embeddedpixelsize = args.embeddedpixelsize
+    logger.info('embeddedpixelsize = {}'.format(embeddedpixelsize))
+    
+    #Required units for the features extracted
+    unitLength = args.unitLength
+    logger.info('unitLength = {}'.format(unitLength))
+    
+    #Pixels per unit vaue for the units mentined in unitLength
+    pixelsPerunit = args.pixelsPerunit
+    logger.info('pixels per unit = {}'.format(pixelsPerunit))
+    
+    #Path to intensity image directory
+    intDir = args.intDir
+    logger.info('intDir = {}'.format(intDir))
+    
+    #Pixel distance to calculate neighbors
+    pixelDistance = args.pixelDistance
+    logger.info('pixelDistance = {}'.format(pixelDistance))
+    
+    #Path to labeled image directory
+    segDir = args.segDir
+    logger.info('segDir = {}'.format(segDir))
```

Comment:
For both `intDir` and `segDir`, you need to check for an `images` subdirectory, and if it is present the plugin needs to switch to that directory.